tests/test_utils.py

```diff
@@ -32,3 +32,18 @@ def test_check_embeddings_shape():
     embeddings = np.array([[1, 2, 3],
                            [2, 3, 4]])
     check_embeddings_shape(embeddings, docs)
+
+
+def test_make_unique_distances():
+    def check_dists(dists: list[float], noise_max: float):
+        unique_dists = get_unique_distances(np.array(dists, dtype=float), noise_max=noise_max)
+        assert len(unique_dists) == len(dists), "The number of elements must be the same"
+        assert len(dists) == len(np.unique(unique_dists)), "The distances must be unique"
+
+    check_dists([0, 0, 0.5, 0.75, 1, 1], noise_max=1e-7)
```

Comment:
Have you checked the actual values of the updated distance list? When I run it, I get the following updated values:

```python
[0.00000000e+00 8.32483552e-08 5.00000000e-01 7.50000000e-01
 1.00000000e+00 2.00000008e+00]
```

The last value is twice as big which should not happen. I have a feeling the code for `get_unique_distances` could be simplified a bit. What about simply doing something like this:

```python
def get_unique_distances(dists):
    increment =  np.random.uniform(low=1e-5, high=1e-6)
    last_val = -float('inf')
    return [last_val := max(dist, last_val + increment) for dist in dists]

my_list = [0, 0, 0, 0.5, 0.75, 1, 1]
get_unique_distances(my_list)
```

Reply:
> Are we ok with changing distances that do not have a duplicate?

Hmmm, my preference would indeed be to keep them as is as long as it requires no more than one or two lines of code. I would like to simplify this as much as possible.

Reply:
I simplified the code. Please have a look and let me know if you have any ideas.

Reply:
Thanks for the changes! I just tested it a bunch of times and it all looks good to me. Thanks for simplifying the code. I'll re-run the workflow to check whether everything passes. If it does, I will go ahead and merge the PR.

Reply:
The tests failed but I believe because you used `list[float]` which is not supported in python 3.8. Removing that should make the tests pass I think.